src/oc/nrnran123.cpp

```diff
@@ -28,6 +28,13 @@ uint32_t nrnran123_get_globalindex() {
 nrnran123_State* nrnran123_newstream(uint32_t id1, uint32_t id2) {
     return nrnran123_newstream3(id1, id2, 0);
 }
+
+nrnran123_State* nrnran123_newstream0() {
+    extern int nrnmpi_myid;
+    static uint32_t id3{};
```

Comment:
By the way, why are we using different function names everywhere here? Can we not just use function overloading? If you don't want to change this for the existing functions, at least the newly introduced ones should be overloaded.

Reply:
No "good" reason.  I'm close to pleading exhaustion with this PR. I think the name variants exist because at implementation I took a sideways glance at their usefulness for a C API. In some ways I think the simplification process could best (and more completely) be done in a future PR since  simplification of the nrnran123.h API could be thought of as being outside the principle domain of this PR.